### PR TITLE
add completed task 4

### PR DESCRIPTION
**Pull request by GemboJones**
completed task 4. I would like some feedback on improving my code for the 404 test if possible as I know this isnt quite right. Thanks!

### CODE REVIEW

File: controllers/app.controller.js [hunk 1]
```diff
@@ -13,5 +13,15 @@ const getTopics = (request, response) => {
     })
 }
 
+const getArticles = (request, response, next) => {
+    const { article_id } = request.params
+    readArticles(article_id)
```
Comment (dappernerddesigns) [inline controllers/app.controller.js]: Minor picky comment about the function name, does this function read articles? Or does it fetch data? getArticles is a good name and you can't use it twice, so a synonym of get would be great, fetchArticles is a good alternative.

File: models/app.model.js [hunk 2]
```diff
@@ -7,4 +7,15 @@ const readTopics = () => {
     })
 }
 
-module.exports = { readTopics }
\ No newline at end of file
+const readArticles = (article_id) => {
+    return db.query(`SELECT * FROM articles WHERE article_id = $1`, [article_id])
+    .then(({rows}) => {
+        if (rows.length === 0) {
+            return Promise.reject({ status: 404, msg: 'not found' })
```
Comment (dappernerddesigns) [inline models/app.model.js:14]: The Promise reject is perfect, exactly what I'd be looking to check.

File: app.js [hunk 3]
```diff
@@ -1,12 +1,31 @@
...
+    } else {
+        next(err)
+    }
+})
+
+app.use((err, request, response, next) => {
+    if (err.status === 404) {
+        response.status(404).send({msg: 'not found'})
+    } else {
+        next(err)
```
Comment (dappernerddesigns) [inline app.js:27]: This is extremely close. Best advice I can give you is to look at the bottom of the error handling notes where there is an example of error functions extracted out into individual handlers. It's very close, I'd move it above the psql handler and consider changing the if statement to something like ```if(err.status && err.msg)``` That way this custom handler is more reusable.

**Review by dappernerddesigns — APPROVED** [review]
A great effort on this one. Some minor bits I'd like you to look at before you move on. No need to resubmit.

File: controllers/app.controller.js (hunk 1 above)
Comment (GemboJones) [reply controllers/app.controller.js]: @dappernerddesigns hope I'm replying to this comment correctly! in that case would you say the name for 'readTopics' function should also be changed to 'fetchTopics'? as I figure both functions are 'fetching' data? I'm unsure of the difference. Appreciate the specifics as I'm a perfectionist so like to get things as correct as poss! :)

**Comment by GemboJones** [comment]
> A great effort on this one. Some minor bits I'd like you to look at before you move on. No need to resubmit.

in that case would I make the changes and then add/commit/push? 
*edit* just realised that you meant you don't need to check my changes, thanks.

File: app.js (hunk 3 above)
Comment (GemboJones) [reply app.js:27]: changed to `app.use((err, request, response, next) => {
    if (err.status && err.msg) {
        response.status(err.status).send({ msg: err.msg });
    } else {
        next(err)
    }
})`

**Comment by GemboJones** [comment]
merging task 4 and 5 on one ticket following feedback from Rayhaan on a zoom call.

**Pull request merged**